Engine/DataFeeds/ZipDataCacheProvider.cs

```diff
@@ -290,12 +334,18 @@ public class CachedZipFile : IDisposable
         /// </summary>
         public string Key { get; private set; }
 
+        /// <summary>
+        /// Returns if this cached zip file is disposed
+        /// </summary>
+        public bool Disposed { get; private set; }
```

Comment:
Any concerns about this backing field _not_ being marked as `volatile`? Perhaps we need a `Thread.MemoryBarrier()` each time we set or retrieve the value to prevent CPU re-ordering of the instructions? Here's a great [stackoverflow](https://stackoverflow.com/a/3556877/1582922) answer discussing the lock-release semantics of .net framework as opposed to the ECMA spec itself (which uses a weaker memory model, and IIRC, mono follows the weaker model leading to bugs). The concern is the 'freshness' of the value, in other words, if ThreadA sets `Disposed=true`, then how long before ThreadB sees that `true` value since it may have cached the value from main memory. The  memory barrier (or also `volatile`) will force the CPU to retrieve the latest value from main memory (if memory serves :) )